vendor/bat-native-ledger/src/bat/ledger/internal/sku/sku_util.cc

```diff
@@ -0,0 +1,153 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <utility>
+
+#include "base/json/json_reader.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/values.h"
+#include "bat/ledger/global_constants.h"
+#include "bat/ledger/internal/sku/sku_util.h"
+#include "bat/ledger/internal/uphold/uphold_util.h"
+
+namespace braveledger_sku {
+
+const char kAnonCardDestinationDev[] =
+    "9094c3f2-b3ae-438f-bd59-92aaad92de5c";
+const char kAnonCardDestinationStaging[] =
+    "6654ecb0-6079-4f6c-ba58-791cc890a561";
+const char kAnonCardDestinationProduction[] =
+    "4cad865e-fea3-486c-a0d6-07c1f187e282";
+
+ledger::SKUOrderPtr ParseOrderCreateResponse(
+    const std::string& response,
+    const std::vector<ledger::SKUOrderItem>& order_items) {
+  base::Optional<base::Value> dictionary = base::JSONReader::Read(response);
+  if (!dictionary || !dictionary->is_dict()) {
+    return nullptr;
+  }
+
+  auto order = ledger::SKUOrder::New();
+
+  const auto* id = dictionary->FindStringKey("id");
+  if (id) {
+    order->order_id = *id;
+  }
+
+  if (order->order_id.empty()) {
+    return nullptr;
+  }
+
+  const auto* total_amount = dictionary->FindStringKey("totalPrice");
+  if (total_amount) {
+    order->total_amount = std::stod(*total_amount);
+  }
+
+  const auto* merchant_id = dictionary->FindStringKey("merchantId");
+  if (merchant_id) {
+    order->merchant_id = *merchant_id;
+  }
+
+  const auto* location = dictionary->FindStringKey("location");
+  if (location) {
+    order->location = *location;
+  }
+
+  order->status = ledger::SKUOrderStatus::PENDING;
+
+  auto* items = dictionary->FindListKey("items");
+  if (items) {
+    if (items->GetList().size() != order_items.size()) {
+      return nullptr;
+    }
+
+    int count = 0;
+    for (auto& item : items->GetList()) {
+      auto order_item = ledger::SKUOrderItem::New();
+      order_item->order_id = order->order_id;
+      order_item->sku = order_items[count].sku;
+      order_item->type = order_items[count].type;
+
+      const auto* id = item.FindStringKey("id");
+      if (id) {
+        order_item->order_item_id = *id;
+      }
+
+      const auto quantity = item.FindIntKey("quantity");
+      if (quantity) {
+        order_item->quantity = *quantity;
+      }
+
+      const auto* price = item.FindStringKey("price");
+      if (price) {
+        base::StringToDouble(*price, &order_item->price);
```

Comment:
`StringToDouble` returns false in case of Overflow/Underflow/NaN conditions. Only setting the value when its true makes sense.

Reply:
@jumde I think if it is false it will just set it to 0 and because this is new object it will not override anything. Do you think that we need to add additional if check?

Reply:
@NejcZdovc - Its a best case conversion. If there are whitespaces or ambiguous characters in the string, the value will be incorrect. More details here: https://cs.chromium.org/chromium/src/base/strings/string_number_conversions.h?q=StringToDouble&sq=package:chromium&dr=CSs&l=62

Reply:
make sense, can change it

Reply:
added